xapian-letor/ranker/listnet_ranker.cc

```diff
@@ -151,7 +153,7 @@ ListNETRanker::save_model_to_metadata(const string & model_key) {
     }
     ostringstream oss;
     for (size_t i = 0; i <  parameters.size(); ++i)
-	oss << setprecision(numeric_limits<double>::digits10) << parameters[i] << endl;
+	oss << serialise_double(parameters[i]) << endl;
```

Comment:
You really just need to concatenate the output of `serialise_double()` into a string.  No need to the `ostringstream` (which is terribly slow IME) and no need to put in newlines.

```
string model_data;
for (size_t i = 0; i < parameters.size(); ++i)
    model_data += serialise_double(parameters[i];
```